src/wp-admin/includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,634 @@
+<?php
+/**
+ * Dependencies manager for plugins.
+ *
+ * @package dependencies-manager.
+ * @since 1.0
+ */
+
+/**
+ * Plugins dependencies manager.
+ *
+ * @since 1.0.0
+ */
+class WP_Plugin_Dependencies {
+
+	/**
+	 * The database option where we store the array of plugins that should be active
+	 * but are not due to unmet dependencies.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 *
+	 * @var string
+	 */
+	protected $pending_plugin_activations_option = 'pending_plugin_activations';
```

Comment:
Since this is just a string denoting the option name, wouldn't using a class constant be better?

```php
const PENDING_PLUGIN_ACTIVATIONS_OPTION = 'pending_plugin_activations';
```

I haven't used class visibility modifiers, because that is a PHP 7.1 feature.